cpp/src/DataStorm/CallbackExecutor.cpp

```diff
@@ -32,7 +35,15 @@ CallbackExecutor::CallbackExecutor() : _flush(false), _destroyed(false)
                 {
                     try
                     {
-                        p.second();
+                        if (_callbackExecutor)
+                        {
+                            // TODO do we need to ensure p.second is executed before we continue?
```

Comment:
I think this need to at least ensure that p.first is kept alive.

Reply:
What is the relationship between p.first and p.second. And if the p.second function needs p.first to remain alive, should we pass p.first as an argument to p.second?

Reply:
I'm reworking this in a separate PR to only keep the callbacks.